File: apis/playlists-service/internal/repositories/playlistsRepository/interface.go
package playlistsRepository

import e "playlists-service/internal/entities"

type IPlaylistsRepository interface {
	FindById(request FindPlaylistByIdRequest) (*e.Playlist, error)
	FindByName(request FindPlaylistByNameRequest) (*e.Playlist, error)
	FindAll(request FindAllPlaylistsRequest) ([]e.Playlist, error)
	FindByIdWithTracks(request FindPlaylistByIdRequest) (*e.Playlist, error)
	Store(request StorePlaylistRequest) error
	Update(request UpdatePlaylistRequest) error
	Delete(request DeletePlaylistRequest) error

	FindTrack(request FindTrackRequest) (*e.Track, error)
	StoreTrack(request StoreTrackRequest) error
	UpdateTrack(request UpdateTrackRequest) error
	DeleteTrack(request DeleteTrackRequest) error
}

File: apis/musics-service/src/repositories/MusicsRepository/interface.ts
import { PaginationRequest } from './dtos';
import Music from '@entities/Music';

export default interface IMusicsRepository {
  find(id: string): Promise<Music | undefined>;
  findAll(request: PaginationRequest): Promise<Music[]>;
  store(music: Music): Promise<void>;
  update(music: Music): Promise<void>;
  delete(id: string): Promise<void>;
  findMostViewed(request: PaginationRequest): Promise<Music[]>;
}
